src/main/java/dataStructures/trie/README.md

```diff
@@ -0,0 +1,73 @@
+# Trie
+
+## Background
+A trie (pronounced as 'try') also known as a prefix tree, is often used for handling textual data, especially in 
+scenarios involving prefixes. In fact, the term 'trie' comes from the word 'retrieval'.
+
+Like most trees, a trie is composed of nodes and edges. But, unlike binary trees, its node can have more than 
+2 children. A trie stores words by breaking down into characters and organising these characters within a hierarchical 
+tree. Each node represents a single character, except the root, which does not represent any character 
+but acts as a starting point for all the words stored. A path in the trie, which is a sequence of connected nodes 
+from the root, represents a prefix or a whole word. Shared prefixes of different words are represented by common paths. 
+To distinguish complete words from prefixes within the trie, nodes are often implemented with a boolean flag. 
+This flag is set to true for nodes that correspond to the final character of a complete word and false otherwise.
```

Comment:
Think this should be highlighted/not lumped together, since it is an important implementation detail.